cunumeric/eager.py

```diff
@@ -680,7 +712,47 @@ def unary_reduction(
                 stacklevel=(stacklevel + 1),
             )
             return
-        if op == UnaryRedCode.ARGMAX:
+        if op == UnaryRedCode.ALL:
+            try:
+                # Try the new version of this interface for NumPy
+                np.all(
+                    rhs.array,
+                    out=self.array,
+                    axis=axes,
+                    keepdims=keepdims,
+                    where=where
+                    if not isinstance(where, EagerArray)
+                    else where.array,
+                )
+            except Exception:
+                np.all(
+                    rhs.array,
+                    out=self.array,
+                    axis=axes,
+                    keepdims=keepdims,
```

Comment:
I believe this block is meant for older versions of `numpy`, that didn't support `keepdims`. I actually don't like this approach for multiple reasons. I would say get rid of the "Exception" path altogether in new code, and I will start a conversation about refactoring the old code as well.

Reply:
OK. Should Start this conversation as a GitHub issue or just via slack?

Reply:
Let's see what comments the Slack conversation attracts, then we can potentially bring it up in a meeting.